packages/core/src/renderer/components/+namespaces/namespace-tree-view.test.tsx

```diff
@@ -0,0 +1,334 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import type { DiContainer } from "@ogre-tools/injectable";
+import { fireEvent } from "@testing-library/react";
+import React from "react";
+import { Namespace } from "../../../common/k8s-api/endpoints";
+import { getDiForUnitTesting } from "../../getDiForUnitTesting";
+import type { DiRender } from "../test-utils/renderFor";
+import { renderFor } from "../test-utils/renderFor";
+import hierarchicalNamespacesInjectable from "./hierarchical-namespaces.injectable";
+import { NamespaceTreeView } from "./namespace-tree-view";
+import type { NamespaceTree } from "./store";
+
+jest.mock("react-router-dom", () => ({
+  Link: ({ children }: { children: React.ReactNode }) => children,
+}));
+
+function createNamespace(name: string, labels?: Record<string, string>, annotations?: Record<string, string>): Namespace {
+  return new Namespace({
+    apiVersion: "v1",
+    kind: "Namespace",
+    metadata: {
+      name,
+      resourceVersion: "1",
+      selfLink: `/api/v1/namespaces/${name}`,
+      uid: `${name}`,
+      labels: {
+        ...labels,
+      },
+      annotations: {
+        ...annotations,
+      },
+    },
+  });
+}
```

Comment:
So why did you still chosen using `Namespace` object for `<NamespaceTreeView root={objectTreeItem}/>`?
Do we need something more than just a `namespace.getName()` from the `Namespace`-object (besides getting access to labels, etc)?

So I did suggested to to use for `objectTreeItem` only plain data types, so the exported tree should  be something like:

```ts
type TreeItem = {namespace: string, expanded?: boolean; children?: TreeItem[] }
```

In that case benefits of it:

- it's already exportable/serializable json format in case of persisting data or sending over network
- simplifies your test with the component so you just operate by namespace names there
- simplifies component itself (NamespaceTreeView) and renders plain data tree

Converting namespace to json-tree we can assign to `NamespaceStore` probably or some separated function-helper that could be easily unit-tested. 
Could be done in separated follow-up PR too. Wdyt?

Reply:
Make sense, will do the refactoring.